File: responser-backend/src/main/java/com/responser/backend/controller/payload/ApiError.java
package com.responser.backend.controller.payload;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * ApiError
 *
 * @author SIE
 */
@NoArgsConstructor
@Data
public class ApiError<T> {

    private Boolean apiError = true;

    private String message;

    private ApiErrorType errorType;

    private T data;

    public ApiError(String message,ApiErrorType errorType) {
        this(message, errorType, null);
    }

    public ApiError(String message, ApiErrorType errorType, T data) {
        this.message = message;
        this.errorType = errorType;
        this.data = data;
    }
}

File: reviewly-chrome-extension/src/app/model/ApiError.ts
export class ApiError extends Error {

    apiError: boolean;

    message: string;

    errorType: ApiErrorType;

    data: any;

    constructor(apiError: boolean, message: string, errorType: ApiErrorType, data: any) {
        super(message);
        this.apiError = apiError;
        this.message = message;
        this.errorType = errorType;
        this.data = data;
    }
}

export enum ApiErrorType {

    ENTITY_NOT_FOUND = "ENTITY_NOT_FOUND",

    VALIDATION_ERROR = "VALIDATION_ERROR"
}
